File: BackEnd/gamedex/src/main/java/cat/copernic/gamedex/entity/Library.java
package cat.copernic.gamedex.entity;

import java.time.LocalDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Field;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

/**
 * Classe que representa una biblioteca de videojocs.
 */
@Document(collection = "library")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Library {

    /**
     * L'identificador de la biblioteca.
     */
    @Id
    @Field(name = "idLibrary")
    private String idLibrary;

    /**
     * L'estat del videojoc a la biblioteca.
     */
    private StateType state;

    /**
     * La descripció del videojoc a la biblioteca.
     */
    private String description;

    /**
     * La puntuació del videojoc a la biblioteca.
     */
    private double rating;

    /**
     * L'usuari propietari de la biblioteca.
     */
    @DBRef
    private User user;

    /**
     * El videojoc a la biblioteca.
     */
    @DBRef
    private Videogame videogame;

    /**
     * La data de publicació del videojoc a la biblioteca.
     */
    @Field(name = "published_date")
    private LocalDate publishedDate;
}

File: FrontEnd/app/src/main/java/cat/copernic/grup4/gamedex/Core/Model/Library.kt
package cat.copernic.grup4.gamedex.Core.Model

/**
 * Classe que representa una biblioteca de videojocs.
 *
 * @param idLibrary L'identificador de la biblioteca.
 * @param user L'usuari propietari de la biblioteca.
 * @param videogame El videojoc a la biblioteca.
 * @param state L'estat del videojoc a la biblioteca.
 * @param description La descripció del videojoc a la biblioteca.
 * @param rating La puntuació del videojoc a la biblioteca.
 * @param publishedDate La data de publicació del videojoc a la biblioteca.
 */
data class Library(
    val idLibrary: String,
    val user: User,
    val videogame: Videogame,
    val state: StateType,
    val description: String,
    val rating: Double,
    val publishedDate: String
)